src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs

```diff
@@ -102,15 +140,25 @@ public Result OnShutdown(UIControlledApplication application)
 
     [Transaction(TransactionMode.Automatic)]
     [Regeneration(RegenerationOption.Manual)]
-    public class Start06x : IExternalCommand
+    public class Command0 : IExternalCommand
     {
         public Result Execute(ExternalCommandData commandData, ref string message, ElementSet elements)
         {
-            var loadPath = Path.Combine(VersionLoader.BasePath, "DynamoRevit.dll");
+            if (!VersionLoader.CacheApplicationPath(0, commandData.Application.Application.VersionNumber))
+                return Result.Failed;
 
-            Utils.WriteToFile(loadPath, commandData.Application.Application.VersionName);
+            return Result.Succeeded;
+        }
+    }
 
-            Utils.ShowRestartMessage(FileVersionInfo.GetVersionInfo(loadPath).FileVersion);
+    [Transaction(TransactionMode.Automatic)]
+    [Regeneration(RegenerationOption.Manual)]
+    public class Command1 : IExternalCommand
```

Comment:
Is not there a better way to do this? It seems like you are creating `Command`s for each product. Can we not dynamically create as many `Command` classes as there are products?